src/configure/clients/azure/appServiceClient.ts

```diff
@@ -36,4 +42,81 @@ export class AppServiceClient extends AzureResourceClient {
 
         return resourceList;
     }
+
+    public async getDeploymentCenterUrl(resourceId: string): Promise<string> {
+        let deploymentCenterUrl = `${this.portalUrl}/#@${this.tenantId}/resource/${resourceId}/vstscd`;
+        return deploymentCenterUrl;
+    }
+
+    public async getVstsPipelineUrl(resourceId: string): Promise<string> {
+        let metadata = await this.getAppServiceMetadata(resourceId);
+        if (metadata.properties['VSTSRM_BuildDefinitionWebAccessUrl']) {
+            return metadata.properties['VSTSRM_BuildDefinitionWebAccessUrl'];
+        }
+
+        throw new Error(Messages.cannotFindPipelineUrlInMetaDataException);
+    }
+
+    public async getAppServiceConfig(resourceId: string): Promise<SiteConfigResource> {
+        let parsedResourceId: ParsedAzureResourceId = new ParsedAzureResourceId(resourceId);
+        return this.webSiteManagementClient.webApps.getConfiguration(parsedResourceId.resourceGroup, parsedResourceId.resourceName);
+    }
+
+    public async updateScmType(resourceId: string): Promise<SiteConfigResource> {
+        let siteConfig = await this.getAppServiceConfig(resourceId);
+        siteConfig.scmType = ScmTypes.VSTSRM;
+        let parsedResourceId: ParsedAzureResourceId = new ParsedAzureResourceId(resourceId);
+        return this.webSiteManagementClient.webApps.updateConfiguration(parsedResourceId.resourceGroup, parsedResourceId.resourceName, siteConfig);
+    }
+
+    public async getAppServiceMetadata(resourceId: string): Promise<StringDictionary> {
+        let parsedResourceId: ParsedAzureResourceId = new ParsedAzureResourceId(resourceId);
+        return this.webSiteManagementClient.webApps.listMetadata(parsedResourceId.resourceGroup, parsedResourceId.resourceName);
+    }
+
+    public async updateAppServiceMetadata(resourceId: string, metadata: StringDictionary): Promise<StringDictionary> {
+        let parsedResourceId: ParsedAzureResourceId = new ParsedAzureResourceId(resourceId);
+        return this.webSiteManagementClient.webApps.updateMetadata(parsedResourceId.resourceGroup, parsedResourceId.resourceName, metadata);
+    }
+
+    public async publishDeploymentToAppService(resourceId: string, buildDefinitionUrl: string, releaseDefinitionUrl: string, triggeredBuildUrl: string): Promise<Deployment> {
```

Comment:
we can release dinifition url as input altogether as we always configure one pipeline. This is in sync with deploy task being used in build where there will be no release